zebra-consensus/src/transaction.rs

```diff
@@ -396,6 +398,26 @@ where
                 )?,
             };
 
+            if let Some(unmined_tx) = req.mempool_transaction() {
+                let check_anchors_and_revealed_nullifiers_query = state
+                    .clone()
+                    .oneshot(zs::Request::CheckBestChainTipNullifiersAndAnchors(
+                        unmined_tx,
+                    ))
+                    .map(|res| {
+                        let res: Result<zs::Response, TransactionError> = res.map_err(|err| {
+                            let err: ValidateContextError = *err.downcast()
+                                .expect("CheckBestChainTipNullifiersAndAnchors error must be a ValidateContextError");
+                            err.into()
+                        });
+                        assert!(res? == zs::Response::ValidBestChainTipNullifiersAndAnchors);
```

Comment:
If we get this panic, it helps to explain why it happened.

```suggestion
                        assert!(res? == zs::Response::ValidBestChainTipNullifiersAndAnchors, "unexpected response to CheckBestChainTipNullifiersAndAnchors request");
```